Add SpoilerOpened tracking event, for saving in PlanetScale

shn-srl opened the pull request:
Sends a SpoilerOpened tracking event to Planetscale

To confirm:

[ ] Is all the required information being saved ?

- Important: PlanetScale migration also needs to run

https://app.planetscale.com/serlo/serlo_planetscale/test-spoiler-tracking/

![image](https://github.com/serlo/frontend/assets/161819227/0e8ac080-43e1-4f3e-819e-709b574f8c71)

shn-srl committed:
Add SpoilerOpened tracking event, for saving in PlanetScale

shn-srl committed:
Fix lint problems

shn-srl committed:
Use AB Tracking for tracking spoiler open events

shn-srl commented on packages/editor/src/plugins/spoiler/static.tsx:
```diff
@@ -1,18 +1,40 @@
+    if(!ab || hasSentSpoilerTrackingEvent) return
+    // send tracking event
+    abSubmission({
+      entityId: -1, // should this be a real id?
```
This needs to be double checked.

If entity id is needed, what id is it? I see string Ids but the abSubmission expects an int

shn-srl committed:
Use ExerciseSubmission

shn-srl committed:
Prettier

Entkenntnis reviewed (changes requested):
Two minor things:

- Can you revert the db change?
- Check if entity id is one of the three we use this event for

shn-srl committed:
Only track spoiler open events for specific ids

shn-srl commented:
> Two minor things:
> 
>     * Can you revert the db change?
> 
>     * Check if entity id is one of the three we use this event for

DB change was on a development branch, which is now deleted :heavy_check_mark: 

Added EntityId check :heavy_check_mark:

Entkenntnis reviewed (changes requested):
move code to integration and out of editor repo

shn-srl committed:
Move spoiler tracking logic to new plugin-wrapper component SpoilerSerloStaticRenderer

shn-srl committed:
Lint

shn-srl committed:
Fix schema

Entkenntnis reviewed (approved):
Thank you
